### PR TITLE
feat(hive): support prewhere in hive

### PR DESCRIPTION
**Pull request by sandflee**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
as ParquetTableSource is still in experimental stage, let's first support prewhere in hive

part  #9347

### CODE REVIEW

**Review by dantengsky — APPROVED** [review]

File: src/query/storages/hive/hive/src/hive_table.rs [hunk 1]
```diff
@@ -255,29 +281,84 @@ impl HiveTable {
...
 
-        let hive_partition_filler = if !partition_fields.is_empty() {
-            Some(HivePartitionFiller::create(partition_fields))
-        } else {
-            None
-        };
+    // Build the prewhere filter executor.
+    fn build_prewhere_filter_executor(
+        &self,
+        _ctx: Arc<dyn TableContext>,
```
Comment (RinChanNOWWW) [inline src/query/storages/hive/hive/src/hive_table.rs:319]: Can be removed.

File: src/query/storages/hive/hive/src/hive_table_source.rs [hunk 2]
```diff
@@ -36,22 +41,31 @@ use crate::HiveBlockFilter;
...
+
+    PrewhereFilter(HiveBlocks, DataBlockDeserializer),
 
     /// Deserialize block from the given data groups
     /// CPU bound
-    Deserialize(HiveBlocks, DataBlockDeserializer),
+    Deserialize(HiveBlocks, DataBlockDeserializer, PreWhereData),
 
     /// `(_, _, Some(_))` indicates that a data block is ready, and needs to be consumed
     /// `(_, _, None)` indicates that there are no more blocks left for the current row group of `HiveBlocks`
```
Comment (RinChanNOWWW) [inline src/query/storages/hive/hive/src/hive_table_source.rs:67]: These comments can be changed.

File: src/query/storages/hive/hive/src/hive_table.rs (hunk 1 above)
Comment (sandflee) [reply src/query/storages/hive/hive/src/hive_table.rs]: done

File: src/query/storages/hive/hive/src/hive_table_source.rs [hunk 3]
```diff
@@ -133,26 +273,28 @@ impl Processor for HiveTableSource {
...
-                        bytes: data_block.memory_size(),
-                    };
-                    self.scan_progress.incr(&progress_values);
+                // 1. consume all generated blocks,
+                if let Some(data_block) = data_blocks.pop() {
+                    // let progress_values = ProgressValues {
+                    // rows: data_block.num_rows(),
+                    // bytes: data_block.memory_size(),
+                    // };
+                    // self.scan_progress.incr(&progress_values);
```
[the remark this answers is not in the document]
Comment (sandflee) [reply src/query/storages/hive/hive/src/hive_table_source.rs]: done

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**